src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/Documents/VisualStudioEditorDocumentManager.cs

```diff
@@ -52,8 +53,12 @@ internal class VisualStudioEditorDocumentManager : EditorDocumentManagerBase
             _runningDocumentTable = (IVsRunningDocumentTable4)runningDocumentTable;
             _editorAdaptersFactory = editorAdaptersFactory;
 
-            var hr = runningDocumentTable.AdviseRunningDocTableEvents(new RunningDocumentTableEventSink(this), out _);
-            Marshal.ThrowExceptionForHR(hr);
+            joinableTaskContext.Factory.Run(() =>
+            {
```

Comment:
I was expecting to see an await jtf.SwitchToMainThreadAsync() here

Reply:
> I was expecting to see an await jtf.SwitchToMainThreadAsync() here

Woooops! I checked inthe code I used to try and repro the issue 😆. Thank you!